src/NuGet.Server/Infrastructure/ServerPackageRepository.cs

```diff
@@ -539,6 +542,7 @@ public void ClearCache()
 
                 _serverPackageStore.Clear();
                 _serverPackageStore.Persist();
+                _needsRebuild = true;
```

Comment:
I wasn't able to find it in that code, they look like different locks. But I'm probably missing something